src/uct/sm/mm/mm_iface.c

```diff
@@ -56,8 +57,11 @@ static ucs_status_t uct_mm_iface_get_address(uct_iface_t *tl_iface,
 static int uct_mm_iface_is_reachable(uct_iface_t *tl_iface,
                                      const struct sockaddr *addr)
 {
+    uct_mm_iface_t *iface = ucs_derived_of(tl_iface, uct_mm_iface_t);
+
     return (addr->sa_family == UCT_AF_PROCESS) &&
-           (((uct_sockaddr_process_t*)addr)->node_guid == ucs_machine_guid());
+           (((uct_sockaddr_process_t*)addr)->node_guid ==
+           (ucs_machine_guid() * ucs_string_to_id(iface->super.pd->component->name)));
```

Comment:
sure, i'll add a comment in the code.
it's for preventing mmap and sysv from having the same address for the proc, and therefore prevent them from connecting one to another.